phd-tests/framework/src/guest_os/windows_server_2022.rs

```diff
@@ -0,0 +1,67 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Guest OS adaptations for Windows Server 2022 images. This adapter assumes
+//! the guest OS image has been generalized, that Cygwin has been installed, and
+//! that the local administrator user was configured at image generation time to
+//! have the appropriate password.
+
+use super::{CommandSequence, CommandSequenceEntry, GuestOs};
+
+pub(super) struct WindowsServer2022;
+
+impl GuestOs for WindowsServer2022 {
+    fn get_login_sequence(&self) -> CommandSequence {
+        CommandSequence(vec![
+            // The image is generalized and needs to be specialized. Let it boot
+            // once, then wait for it to boot again and only then start waiting
+            // for the command prompt to become available.
+            CommandSequenceEntry::WaitFor(
+                "Computer is booting, SAC started and initialized.",
+            ),
+            CommandSequenceEntry::WaitFor(
+                "Computer is booting, SAC started and initialized.",
+            ),
+            CommandSequenceEntry::WaitFor(
+                "EVENT: The CMD command is now available.",
+            ),
+            CommandSequenceEntry::WaitFor("SAC>"),
+            CommandSequenceEntry::WriteStr("cmd"),
+            CommandSequenceEntry::WaitFor("Channel: Cmd0001"),
+            CommandSequenceEntry::WaitFor("SAC>"),
+            CommandSequenceEntry::WriteStr("ch -sn Cmd0001"),
+            CommandSequenceEntry::WaitFor(
+                "Use any other key to view this channel.",
+            ),
+            CommandSequenceEntry::WriteStr(""),
+            CommandSequenceEntry::WaitFor("Username:"),
+            CommandSequenceEntry::WriteStr("Administrator"),
+            CommandSequenceEntry::WaitFor("Domain  :"),
+            CommandSequenceEntry::WriteStr(""),
+            CommandSequenceEntry::WaitFor("Password:"),
+            CommandSequenceEntry::WriteStr("0xide#1Fan"),
```

Comment:
If we're going to be encoding more expectations about image login credentials, it might make sense to have some top-level markdown doc listing them so that folks don't need to go fishing in the source for them?

Reply:
Good call, done in 40aed6a.